Lib/idlelib/ParenClose.py

```diff
@@ -0,0 +1,115 @@
+"""
+Parens and Ticks Closing Extension
+
+When you hit left paren or tick,
+automatically creates the closing paren or tick.
+
+Author: Charles M. Wohlganger
+        charles.wohlganger@gmail.com
+
+Last Updated: 13-Aug-2017 by Charles Wohlganger
+
+Add to the end of config-extensions.def :
+
+    [ParenClose]
+    enable = True
+    paren_close = True
+    tick_close = True
+    skip_closures = True
+    mutual_delete = True
```

Comment:
Changing the specification of the new feature by adding action on deletion should have been discussed on the issue.  (This applies generally to changing specs.) Notepad++ does do anything special on deletion.  Binding the deletion keys here prevents binding them elsewhere, and I have an idea of where it might be needed, as opposed to optional.  I also think sometimes delete more and sometimes not is confusing.  The deletion part of 2f19ca8 and anything later should be reverted unless and until there is a clear agreement on adding this.